### PR TITLE
New - __::isEqual()

### LINKED ISSUE #46: Add a recursive array / object equality __::isEqual
I was searching for a deep equality function for arrays in PHP.

There are [`array_diff`](http://php.net/manual/fr/function.array-diff.php) [`array_deep_assoc`](http://php.net/manual/fr/function.array-diff-assoc.php), but they are not recursive.

If we need inspiration for the algo, see also
* https://github.com/lodash/lodash/blob/4.17.11/lodash.js#L3273
* https://stackoverflow.com/questions/3876435/recursive-array-diff
* https://stackoverflow.com/questions/2937176/does-phpunit-have-some-inbuilt-recursive-array-comparison-function

### PR DESCRIPTION
**Pull request by MonsieurV**
Add an `__::isEqual()` function, that perform deep (recursive) object equality checks.

Work in progress: for now I've just added some acceptance tests.

Closes #46

### CODE REVIEW

**Comment by MonsieurV** [comment]
May merit more tests, but I think it is already useful as it is. (We can't simply get this behavior by directly using `===` nor `==`)

I'll merge it tomorrow if no objections. Feedbacks welcome.

File: src/__/load.php [hunk 1]
```diff
@@ -77,6 +77,7 @@
  * @method static bool isNumber(mixed $value) <p>Check if give value is number or not.</p>
  * @method static bool isObject(mixed $value) <p>Check if give value is object or not.</p>
  * @method static bool isString(mixed $value) <p>Check if give value is string or not.</p>
+ * @method static bool isEqual(mixed $value, mixed $value) <p>Check if the two values are deeply equal or not.</p>
```
Comment (allejo) [inline src/__/load.php]: Might need to run the doc script again, the doc string for this doesn't match the updated phpdocs

Comment (MonsieurV) [reply src/__/load.php]: Thx, I'll go RTFM

File: README.md [hunk 2]
```diff
@@ -517,6 +517,13 @@ __::isCollection([1, 2, 3]);
 // >> true
 ```
 
+##### [__::isCollection](src/__/objects/isEqual.php)
+Performs a deep comparison between two values to determine if they are equivalent.
+```php
+__::isEqual([1, 1.0]);
```
Comment (allejo) [inline README.md]: I think you might be missing one argument from this example, there's only one array

Comment (allejo) [inline README.md]: Oops... Accidentally deleted my comment. Are these docs meant for `isCollection()` or `isEqual()`? If it's for `isEqual()`, I think there's one argument missing from the function call.

File: src/__/objects/isEqual.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        if (count($object1Keys) !== count($object2Keys)) {
+            return false;
+        }
+        foreach ($object1 as $key1 => $value1) {
+            if (!\__::has($object2, $key1) || !\__::isEqual($value1, \__::get($object2, $key1))) {
+                return false;
+            }
+        }
+        return true;
+    } else {
```
[the remark this answers is not in the document]
Comment (MonsieurV) [reply src/__/objects/isEqual.php]: Agree

File: README.md (hunk 2 above)
Comment (MonsieurV) [reply README.md]: Oh yes, that was bad

**Pull request merged**